### PR TITLE
Strategy

### PR DESCRIPTION
**Pull request by vsharonlynn**
Implemented methods: crossover, getWeight

### CODE REVIEW

File: Strategy.java [hunk 1]
```diff
@@ -1,63 +1,87 @@
...
-
-  //TODO: Implement
-  // Mutate this strategy.
-  public void mutate() {
-  }
+	public double getWeight(int index) {
+		return weights[index];
+	}
+
+	// TODO: Implement
```
Comment (digawp) [inline Strategy.java]: Remove the TODOs

Comment (vsharonlynn) [inline Strategy.java]: Just did it on the commit below :)

File: Strategy.java [hunk 2]
```diff
@@ -1,63 +1,91 @@
...
+
+	/**
+	 * Crossover two parent strategies: this (first parent) and
+	 * param (second parent) to produce two children strategies.
+	 * @param s: second parent strategy
+	 * @return array of strategy with 2 elements: two children strategies
+	 */
+	public Strategy[] crossover(Strategy s) {
+		Random random = new Random();
+		int seed = random.nextInt(1000) % (weights.length - 2) + 1;
```
Comment (digawp) [inline Strategy.java:55]: Don't understand why you need to do this. Why not just `%` by `weights.length - 1`? Or `nextInt(weights.length - 1)`?

Comment (nathanajah) [inline Strategy.java:55]: So that the seed can't be zero?
Anyway shouldn't matter that much.

**Pull request merged**